temp_convert/src/main.rs

```diff
@@ -1,78 +1,91 @@
+            true
+        } else {
+            false
+        }
+    }
 
-    match input_symbol {
-        Temperature::Farenheit => {
-            input_temperature = convert_f_to_c(input_temperature);
-            },
-        Temperature::Celsius => {
-            input_temperature = convert_c_to_f(input_temperature);
-            },
+    fn is_fahrenheit(&self) -> bool {
+        !self.is_celsius()
     }
+}
 
-    input_symbol = input_symbol.convert();
-    
-    println!("Your temperature is {}{}", input_temperature, input_symbol.get_symbol());
+// We'll want to implement the Display trait for our enum,
+// so that we can print it using `{}` and get the temperature nicely formatted
+impl Display for Temperature {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use Temperature::*;
+        match *self {
+            // The `write!()` macro works similarly to the `print!()` macro, except that
+            // it takes a target to write to. In our case this is a mutable reference to a std::fmt::Formatter
+            Fahrenheit(t) => write!(f, "{}°F", t),
+            Celsius(t) => write!(f, "{}°C", t),
+        }
+    }
 }
 
-fn get_temperature() -> f32 {
-    let message = "What is the temperature you want to convert?";
-    let temperature = get_user_input(message);
-    temperature.parse::<f32>()
-        .expect("please give me a number next time :(")
+fn main() {
+    println!("Temperature converter!");
+    let temperature = get_temperature();
+    println!("Your input temperature is {}", temperature);
+    println!("After converting your temperature is {}", temperature.convert_to_new());
 }
 
-fn get_temperature_symbol() -> Temperature {
-    let message = "Is this a temperature in c or f?";
-    let temperature_symbol = get_user_input(message);
+fn get_temperature() -> Temperature {
+    let mut buf = String::new(); // We'll reuse this buffer for all our input
+    get_user_input("What is the temperature you want to convert?\n> ", &mut buf);
+
+    let temp = buf.trim().parse()
+        .expect("please give me a number next time :(");
 
-    match temperature_symbol.deref() {
-        "f" | "F" => Temperature::Farenheit,
-        "c" | "C" => Temperature::Celsius,
+    get_user_input("Is this a temperature in c or f?\n> ", &mut buf);
+    match buf.trim() {
+        "f" | "F" => Temperature::Fahrenheit(temp),
+        "c" | "C" => Temperature::Celsius(temp),
         _ => panic!("Please enter C or F"),
     }
 }
 
-fn get_user_input(message: &str) -> String {
-    let mut user_input = String::new();
-    println!("{}", message);
-    io::stdin().read_line(&mut user_input)
+// Providing a buffer for `read_line()` to read into let's us reuse the buffer and save some allocations
+fn get_user_input(message: &str, buffer: &mut String) {
+    buffer.clear(); // Remove any old content in the buffer
+    
+    // We'll use `print!()` instead of `println!()` so we can have a message that doesn't end with a newline
```

Comment:
This should be more like: `[...] so that we don't automatically append a newline to the message and print it like it was given to us`